### ISSUE TITLE
[Feature Request] Programmatically Dialogs

### ISSUE THREAD

**Issue by robsontenorio** [vuetifyjs/vuetify#3037]
### New Functionality
Is there any plans to implement programmatically dialogs like that?

https://buefy.github.io/#/documentation/dialog
```
this.$dialog.alert('Everything looks fine!')

or

this.$dialog.alert({
                    title: 'Title Alert',
                    message: 'I have a title, a custom button and <b>HTML</b>!',
                    confirmText: 'Cool!'
                })
```

### Improvements
Its very usefull if you have many context dialogs in the same page. Many lines of code saved on `<template>`.

### Bugs or Edge Cases it Helps Avoid
Clean html source.

**Comment by nekosaur** [comment]
I know that we've been hesitant to add programmatic functionality like this previously. Any opinions @vuetifyjs/dev-team ?

Personally I think this could be a user land plugin/add-on

**Comment by johnleider** [comment]
We will have to do some evaluation after v1.1. Currently we have a programmatic snackbar in the PR queue with every intent to integrate. Maybe it is possible to create a factory generator that handles both cases.

**Comment by robsontenorio** [comment]
Any plans to support it officially or any add-ons?

**Comment by HamedFathi** [comment]
@johnleider 

Any good news on this? Dialog service is very useful, please support it.
Thanks for your awesome works :D

**Comment by johnleider** [comment]
Yes, coming as a vuex module in the future. Talked about [here](https://slides.com/johnjleider/vuetify-2/fullscreen)

**Comment by zslusnys** [comment]
Is anyone working on this? Maybe I could help solving this one.

**Comment by natxocc** [comment]
I would like the same for Dialog, snackbar.. etc... Better than creating a new snackbar component for my apps.

**Comment by robsontenorio** [comment]
Hi, maintainers how about this? 

https://www.npmjs.com/package/vuetify-dialog

**Comment by johnleider** [comment]
Thank you for the Feature Request and interest in improving Vuetify. After careful consideration the team has decided that this is not functionality that we are looking to implement at this time.

With that being said, I still have intentions of adding this type of functionality into a vuex module in the future.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Comment by kieuminhcanh** [comment]
Dear @johnleider 

It was 4 years since this topic created. This is my opinion.

Programmatically Dialogs is an essential feature when developing web applications, and providing it as a global setting makes the development process much more convenient. By implementing Programmatically Dialogs at a global level, you can eliminate the need for repetitive code and streamline your workflow.

This feature allows for more efficient and cleaner code management, making it easier for developers to maintain and update their projects. As a result, it not only saves time but also reduces the likelihood of errors and inconsistencies arising from duplicate code.

Incorporating Programmatically Dialogs as a global setting will help developers create more cohesive and adaptable applications, which ultimately leads to a better user experience. By taking this step, you will be providing a valuable tool for developers to create more dynamic and interactive web applications while minimizing the complexity of their code.

In conclusion, the integration of Programmatically Dialogs as a global setting will greatly benefit developers and users alike, making the development process more efficient and the resulting applications more engaging and user-friendly.

**Comment by johnleider** [comment]
Can't this be solved in the community?

**Comment by KaelWD** [comment]
That comment was 100% AI generated.

**Comment by kieuminhcanh** [comment]
@KaelWD Thanks, My English is not very well. I checked it so many times before post. But I think this feature is really important for web development. I used https://github.com/yariksav/vuetify-dialog before and it's really useful. But it was stopped 2 years ago.

**Comment by kieuminhcanh** [comment]
@johnleider, @KaelWD, @robsontenorio   

Based on the requirements and needs of the community, I have created a project to provide toasters, dialogs, alerts, confirms, and prompts. It's also support show component in dialog. 

Source: https://github.com/kieuminhcanh/vuetify-notifier

Demo: https://kieuminhcanh.github.io/vuetify-notifier

**Comment by johnleider** [comment]
> Based on the requirements and needs of the community, I have created a project to provide toasters, dialogs, alerts, confirms, and prompts. It's also support show component in dialog.

Very cool. Can you reach out to me on Discord? https://community.vuetifyjs.com/

**Comment by vapdev** [comment]
I would love this feature. Currently I use a v-dialog for showing modals of success, or fails, after CRUD interactions.
And its such a pain having to insert the dialog component in the templates everywhere, having to set props and changing the v-model.
if I could just "this.openModal("success", "custom message"), would be so much better

**Comment by vsambor** [comment]
Why is this closed if the feature is not available?
I am surprised that this is not by default part of the lib, it is really a crucial thing to have when developing an app.
